scout/server/blueprints/variant/controllers.py

```diff
@@ -246,6 +250,10 @@ def variant(
     panels = default_panels(store, case_obj)
     add_gene_info(store, variant_obj, gene_panels=panels, genome_build=genome_build)
 
+    # Update some case panels info from db and populate it on variant to avoid showing removed panels
+    update_case_panels(store, case_obj)
```

Comment:
If I understand what the code does, this might be a bit dangerous if you invert the order of these 2 functions. Because in that case the case panels will still contain the hidden panels. It could happen when developing, by mistake because we won't remember that the first function called does a filtering that is required in the second..

Reply:
To be clear, there is no filtering happening here until we reach the template, only updating. But I agree there is an order dependence that could be made explicit. The cost of e.g. calling `update_case_panels` from `update_variant_case_panels` would be some extra db calls for the panels for every variant, instead of having it done once for the case, then calling for each variant. For the variant page, that is nothing, but for the variantS it gets a little costly. I can add a line to the `update_variant_case_panels` docstring to note this perhaps?

Reply:
Update them both together?

Reply:
Yes, we could, but to avoid the performance penalty when used in the variantS view, we would have to take a list of variant_obj to it then. And to do that, consume the db iterator a little earlier and store all variant objs during the variantS loop below. It is fine I guess, just costs a little bit of memory. And for the variant feels a little silly to knowingly pass a list of one all the time for the variant. But making the code easy to read and maintain can be worth it. I would go with the docstring here, for a measurable gain in either cpu or memory over code complexity. But it is not like we are super starved for either CPU or RAM, so just insist and I will change it.

Reply:
It is fine with a docstring, but perhaps for good measure add also a comment in the function below (I never read the docstrings..)

Reply:
Hihi, I took that to heart from one of the code analysis tools we used (Sourcery or Code factor). When one is compelled to comment, its most often because better function names, or an additional function with a docstring, is needed. The corollary becomes to read docstrings if in doubt about what the function does. 😸

Reply:
Then the fact I never read docstrings means that we use good names for the functions!